contracts/storage/StakedPutStorage.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.7.2;
+pragma experimental ABIEncoderV2;
+
+import {
+    ReentrancyGuard
+} from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import {
+    OwnableUpgradeable
+} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {
+    ERC20Upgradeable
+} from "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+
+import {IRibbonFactory} from "../interfaces/IRibbonFactory.sol";
+
+contract StakedPutStorageV1 is
+    OwnableUpgradeable,
+    ERC20Upgradeable,
+    ReentrancyGuard
+{
+    struct InstrumentPosition {
+        bool exercised;
+        uint8 putVenueID;
+        uint32 putOptionID;
+        uint256 amount;
+        uint256 putStrikePrice;

```

Comment:
We can remove the `putStrikePrice` variable completely from storage because we don't have to look it up from contract. For the frontend, we can always use The Graph for this.

```
 event PositionCreated(
        address indexed account,
        uint256 indexed positionID,
        uint256 amount,
        uint256 putStrikePrice
    );
```

Reply:
Actually- for simplicity's sake do you want to lookup these values directly from Hegic? That's much better because we're not storing the same exact values `strikePrice` and `expiry` across 2 different protocols.

This means we can remove 2 large uint256's out of storage completely:

```solidity
struct InstrumentPosition {
        bool exercised;
        uint8 putVenueID;
        uint32 putOptionID;
        uint256 amount; // we need to retain this still in case the protocol's option token is fungible
        // REMOVED: uint256 putStrikePrice;
        // REMOVED uint256 expiry;
}
```

Reply:
ah yeah good call. will do